File: backend/app/services/notification_service.py
_bot = None


def set_bot(bot):
    global _bot
    _bot = bot


async def send_to_passenger(user_id: int, message: str):
    if not _bot:
        return
    try:
        from app.database import pool
        async with pool.acquire() as conn:
            row = await conn.fetchrow("SELECT telegram_id FROM users WHERE id=$1", user_id)
        if row:
            await _bot.send_message(row["telegram_id"], message)
    except Exception as e:
        print(f"[bot] send_to_passenger error: {e}")


async def send_to_driver(driver_id: int, message: str):
    if not _bot:
        return
    try:
        from app.database import pool
        async with pool.acquire() as conn:
            row = await conn.fetchrow("SELECT telegram_id FROM drivers WHERE id=$1", driver_id)
        if row:
            await _bot.send_message(row["telegram_id"], message)
    except Exception as e:
        print(f"[bot] send_to_driver error: {e}")

File: backend/src/services/notification.service.js
const db = require('../config/database');
const logger = require('../utils/logger');

let bot;
function setBot(b) { bot = b; }

async function sendToPassenger(userId, { type, message, data }) {
  await db('notifications').insert({ user_id: userId, type, message, data: JSON.stringify(data || {}) }).catch(() => {});
  if (!bot) return;
  try {
    const user = await db('users').where({ id: userId }).first();
    if (user?.telegram_id) await bot.telegram.sendMessage(user.telegram_id, message);
  } catch (err) {
    logger.warn('Bot sendMessage failed:', err.message);
  }
}

async function sendToDriver(driverId, { type, message, data }) {
  await db('notifications').insert({ driver_id: driverId, type, message, data: JSON.stringify(data || {}) }).catch(() => {});
  if (!bot) return;
  try {
    const driver = await db('drivers').where({ id: driverId }).first();
    if (driver?.telegram_id) await bot.telegram.sendMessage(driver.telegram_id, message);
  } catch (err) {
    logger.warn('Bot sendMessage failed:', err.message);
  }
}

module.exports = { setBot, sendToPassenger, sendToDriver };
